File: weknot_android/app/src/main/java/com/example/weknot_android/view/fragment/VideoCallFragment.java
package com.example.weknot_android.view.fragment;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProviders;

import com.example.weknot_android.base.BaseFragment;
import com.example.weknot_android.databinding.VideoCallFragmentBinding;
import com.example.weknot_android.viewmodel.VideoCallViewModel;

import org.jitsi.meet.sdk.JitsiMeet;
import org.jitsi.meet.sdk.JitsiMeetActivity;
import org.jitsi.meet.sdk.JitsiMeetConferenceOptions;

import java.net.MalformedURLException;
import java.net.URL;

public class VideoCallFragment extends BaseFragment<VideoCallFragmentBinding> {

    private VideoCallViewModel videoCallViewModel;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        initViewModel();
        initVideoCall();

        observeVideoCallViewModel();

        clickEvent();
    }

    private void initViewModel() {
        videoCallViewModel = ViewModelProviders.of(this).get(VideoCallViewModel.class);
    }

    private void initVideoCall() {
        URL serverURL;
        try {
            serverURL = new URL("https://meet.jit.si");
        } catch (MalformedURLException e) {
            e.printStackTrace();
            throw new RuntimeException("Invalid server URL!");
        }
        JitsiMeetConferenceOptions defaultOptions
                = new JitsiMeetConferenceOptions.Builder()
                .setServerURL(serverURL)
                .setWelcomePageEnabled(false)
                .build();
        JitsiMeet.setDefaultConferenceOptions(defaultOptions);
    }

    private void observeVideoCallViewModel() {
        videoCallViewModel.getData().observe(this, videoCall -> {
            JitsiMeetConferenceOptions options
                    = new JitsiMeetConferenceOptions.Builder()
                    .setRoom(Integer.toString(videoCall.getRoomIdx()))
                    .build();
            JitsiMeetActivity.launch(getContext(), options);
        });
    }

    private void clickEvent() {
        binding.videoCallBtn.setOnClickListener(v -> videoCallViewModel.requestCall());
    }

    @Override
    protected int layoutId() {
        return 0;
    }
}

File: weknot_android/app/src/main/java/com/example/weknot_android/view/fragment/VideoCallFragment.kt
package com.example.weknot_android.view.fragment

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.lifecycle.Observer
import com.example.weknot_android.BR
import com.example.weknot_android.R
import com.example.weknot_android.base.fragment.BaseFragment
import com.example.weknot_android.databinding.VideoCallFragmentBinding
import com.example.weknot_android.view.activity.ProfileActivity
import com.example.weknot_android.view.activity.VideoCallActivity
import com.example.weknot_android.view.navigator.VideoCallNavigator
import com.example.weknot_android.viewmodel.VideoCallViewModel
import com.facebook.react.modules.core.PermissionListener
import org.jitsi.meet.sdk.*
import org.jitsi.meet.sdk.JitsiMeetConferenceOptions.Builder
import java.net.MalformedURLException
import java.net.URL

class VideoCallFragment : BaseFragment<VideoCallFragmentBinding, VideoCallViewModel>(), VideoCallNavigator {

    override fun getLayoutId(): Int {
        return R.layout.video_call_fragment
    }

    override fun getViewModel(): Class<VideoCallViewModel> {
        return VideoCallViewModel::class.java
    }

    override fun getBindingVariable(): Int {
        return BR.viewModel
    }

    override fun finishEvent() {
        viewModel.finishRoom()
    }

    override fun initObserver() {
        with(viewModel) {
            connectVideoCall.observe(this@VideoCallFragment, Observer {

                val options: JitsiMeetConferenceOptions? = Builder()
                        .setRoom("channel" + it.channel)
                        .setSubject("랜덤채팅")
                        .setUserInfo(JitsiMeetUserInfo(viewModel.userInfo))
                        .build()

                VideoCallActivity.videoLaunch(context!!, options!!, it.channel)
            })

            openProfile.observe(this@VideoCallFragment, Observer {
                val intent = Intent(context, ProfileActivity::class.java)
                intent.putExtra("id", it)
                startActivity(intent)
            })
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        VideoCallActivity.setNavigator(this)
        viewModel.setUp()
    }
}